ocs_ci/ocs/exceptions.py

```diff
@@ -196,3 +196,16 @@ class RDMDiskNotFound(Exception):
 
 class CredReqSecretNotFound(Exception):
     pass
+
+
+class BugInOCSCI(Exception):
```

Comment:
It seems weird to me that we would raise an Error stating that there is a bug in ocs-ci. This is being inherited by `InvalidDeploymentPlatform` which is being raised if the `DEFAULT_STORAGECLASS` attribute isn't being defined. Wouldn't the built-in `NotImplementedError` be a more accurate error to raise?

Reply:
[Exception `NotImplementedError`](https://docs.python.org/3/library/exceptions.html#NotImplementedError) has a very specific meaning:

> This exception is derived from RuntimeError. In user defined base classes, abstract methods should raise this exception when they require derived classes to override the method, or while the class is being developed to indicate that the real implementation still needs to be added.

So using it in this case would be confusing.

What I'm trying to capture here is the fact that every `DeploymentPlatform` class has to have `DEFAULT_STORAGECLASS` defined, and when that is not the case, it's the problem of the particular deployment class itself, and it's author. I want this failure to be very clear.

Now the question is why such failure mode is even considered? This is used in `patch_default_sc_to_non_default()`, which previously just logging that setting SC is not happening in such a case. Which doesn't make sense to me.

Reply:
Having `BugInOCSCI` may seem little weird, I agree, but the idea here is that when internal consistency of ocs-ci or it's assumptions is not met, we want to make it very clear, so that when someone is looking at this during analysis of a CI failure, we immediately see that this is a ocs-ci issue rather than a product bug.

Reply:
I'm going to add a check into init method of  baseclass `Deployment` directly, raising `NotImplementedError` there and removing the BugInOCSCI exceptions.

Reply:
Dropped.